File: mediacodecdemo/src/main/java/com/cfox/mediacodecdemo/MainActivity.java
package com.cfox.mediacodecdemo;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.Manifest;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.os.Environment;
import android.view.SurfaceHolder;
import android.view.SurfaceView;

import java.io.File;

public class MainActivity extends AppCompatActivity {


    public boolean checkPermission() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && checkSelfPermission(
                Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
            requestPermissions(new String[]{
                    Manifest.permission.READ_EXTERNAL_STORAGE,
                    Manifest.permission.WRITE_EXTERNAL_STORAGE
            }, 1);

        }
        return false;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        checkPermission();
        initSurface();

    }

    private void initSurface() {

        SurfaceView surfaceView = findViewById(R.id.sfv);
        final SurfaceHolder surfaceHolder = surfaceView.getHolder();
        surfaceHolder.addCallback(new SurfaceHolder.Callback() {
            @Override
            public void surfaceCreated(@NonNull SurfaceHolder holder) {
//                H264Player h264Player = new H264Player(
//                        new File(Environment.getExternalStorageDirectory(), "out.h264").getAbsolutePath(),
//                        surfaceHolder.getSurface());
//                h264Player.play();

                GetH264Image getH264Image = new GetH264Image(
                        new File(Environment.getExternalStorageDirectory(), "out.h264").getAbsolutePath(),
                        surfaceHolder.getSurface());
                getH264Image.play();
            }

            @Override
            public void surfaceChanged(@NonNull SurfaceHolder holder, int format, int width, int height) {

            }

            @Override
            public void surfaceDestroyed(@NonNull SurfaceHolder holder) {

            }
        });

    }
}

File: CastScreenPlayApp/src/main/java/com/cfox/castscreenplayapp/MainActivity.kt
package com.cfox.castscreenplayapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.SurfaceHolder
import android.view.SurfaceView

class MainActivity : AppCompatActivity() {

    companion object {
        private const val TAG = "PlayMainActivity"
    }

    private val playerManager  = PlayerManager(this)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<SurfaceView>(R.id.sf_play_view).holder.addCallback(object: SurfaceHolder.Callback{
            override fun surfaceChanged(
                holder: SurfaceHolder,
                format: Int,
                width: Int,
                height: Int
            ) {
                Log.d(TAG, "surfaceChanged: =====>")
                val surface = holder.surface
                playerManager.startPlay(surface)

            }

            override fun surfaceDestroyed(holder: SurfaceHolder) {
            }

            override fun surfaceCreated(holder: SurfaceHolder) {
            }

        })
    }
}